--- src/zzuf.c.orig	2016-02-01 23:34:10.584368611 +0000
+++ src/zzuf.c	2016-02-01 23:42:59.512074524 +0000
@@ -722,28 +722,51 @@
         ++slot;
 
     /* Prepare required files, if necessary */
-    if (opts->opmode == OPMODE_COPY)
+    /* Disabled for CERT BFF, as it does the file fuzzing */
+    if (opts->opmode == OPMODE_COPY && 0)
     {
         char tmpname[4096];
         char *tmpdir;
+        char *extension;
+        char *include_env;
+        char *include = "";
         tmpdir = getenv("TEMP");
+       
+        include_env = getenv("ZZUF_INCLUDE");
+        if(include_env)
+	{
+	   include = calloc(strlen(include_env)*sizeof(char), sizeof(char));
+	   strncat(include, include_env+1, strlen(include_env)-2); /* strip out parentheses */
+	}
         if (!tmpdir || !*tmpdir)
             tmpdir = "/tmp";
 
-        int k = 0;
+        int k = 0, extlen;
 
         for (int j = zz_optind + 1; j < opts->oldargc; ++j)
         {
             FILE *fpin = fopen(opts->oldargv[j], "r");
             if (!fpin)
                 continue;
+	   
+	    if(include_env)
+	    {
+		if(strcmp(include, opts->oldargv[j]) !=0)
+		  continue;  /* not included: ignore */		
+	    }
+	    extension = strrchr(opts->oldargv[j], '.');
+	    if (!extension)
+	      extlen=0;
+	    else
+	      extlen=strlen(extension);
+	   
 
 #ifdef _WIN32
             sprintf(tmpname, "%s/zzuf.%i.XXXXXX", tmpdir, GetCurrentProcessId());
             int fdout = _open(mktemp(tmpname), _O_RDWR, 0600);
 #else
-            sprintf(tmpname, "%s/zzuf.%i.XXXXXX", tmpdir, (int)getpid());
-            int fdout = mkstemp(tmpname);
+            sprintf(tmpname, "%s/zzuf.%i.XXXXXX%s", tmpdir, (int)getpid(), extension);
+            int fdout = mkstemps(tmpname, extlen);
 #endif
             if (fdout < 0)
             {
